### PR TITLE
Use xkeysym for keyboard handling

### PR DESCRIPTION
**Pull request by notgull**
Replaces the `keysyms` module with `xkeysym` and makes APIs take `xkeysym::Keysym` instead of alternatives. See notgull/xkeysym#6

### CODE REVIEW

**Comment by kchibisov** [comment]
Could you rebase with the current latest version?

File: Cargo.toml [hunk 1]
```diff
@@ -31,8 +31,9 @@ wayland-protocols-wlr = { version = "0.1.0", features = ["client"] }
 wayland-scanner = "0.30.0"
 wayland-csd-frame = { version = "0.2.2", default-features = false, features = ["wayland-backend_0_1"] }
 
-xkbcommon = { version = "0.5", optional = true, features = ["wayland"] }
+xkbcommon = { version = "0.5.0", optional = true, features = ["wayland"], git = "https://github.com/notgull/xkbcommon-rs.git", branch = "xkeysym" }
 calloop = { version = "0.10.5", optional = true }
+xkeysym = { git = "https://github.com/notgull/xkeysym.git", version = "0.1.1" }
```
Comment (kchibisov) [inline Cargo.toml:36]: It's a bit weird that you've picked it like that, we have a stable release with the good version. And the crate is published.

**Comment by kchibisov** [comment]
And I guess it's not a draft anymore at this point?

**Comment by notgull** [comment]
> And I guess it's not a draft anymore at this point?

Unfortunately I can't take it out of draft status until https://github.com/rust-x-bindings/xkbcommon-rs/pull/43 is released.

**Comment by kchibisov** [comment]
@notgull I mean, we have a release we can use, I don't see any issue with using it here given that it's better than hand rolled solution and it's already used in winit, so we have less code size upstream.

**Comment by notgull** [comment]
Oh, are you talking about `xkbcommon-dl`? Sounds good I'll port to that

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@
 #### Breaking Changes
 
 - `wayland-rs` dependencies are updated to 0.30 and all APIs have changed significantly as a result
+- `xkeysym::Keysym` is used as a keyboard key representation instead of `u32`
```
Comment (kchibisov) [inline CHANGELOG.md:32]: You should put it in the right place.

Comment (notgull) [reply CHANGELOG.md:32]: This is a breaking change, I think?

Comment (kchibisov) [reply CHANGELOG.md:32]: I haven't said that it's not, it just for 0.17 which was released months ago. You should move to the actual unreleased section.

Comment (notgull) [reply CHANGELOG.md]: My bad, fixed!

**Review by PolyMeilex — APPROVED** [review]
Looks great to me.

EDIT:
Looks like it's not possible to use those with match statements?

**Review by wash2 — APPROVED** [review]

**Comment by PolyMeilex** [comment]
Thanks!

**Pull request merged**